threadpoolctl/_threadpool_limiters.py

```diff
@@ -154,14 +155,14 @@ def _set_threadpool_limits(limits=None, user_api=None):
     modules = _load_modules(prefixes=prefixes, user_api=user_api)
     for module in modules:
         n_thread = _get_limit(module['prefix'], module['user_api'], limits)
+        if return_original_limits:
+            module['n_thread'] = module['get_num_threads']()
         if n_thread is not None:
             set_func = module['set_num_threads']
             set_func(n_thread)
 
-        # Store the module and remove un-necessary info
-        module['n_thread'] = module['get_num_threads']()
-        del module['set_num_threads'], module['get_num_threads']
-        del module['clib']
+        if not return_original_limits:
+            module['n_thread'] = module['get_num_threads']()
```

Comment:
You should delete the private part of this module from the report to avoid storing them for nothing. (`'clib'`, `'set_num_threads'` and `'get_num_threads'`).

Reply:
This is private API and its only used in the context manager and the tests, never reported to the user so I think it's fine to always return them.

Reply:
ok